src/bp/core/services/telemetry/telemetry-stats.ts

```diff
@@ -0,0 +1,91 @@
+import axios from 'axios'
+import LicensingService from 'common/licensing-service'
+import { machineUUID } from 'common/stats'
+import { ServerStats } from 'common/telemetry'
+import Database from 'core/database'
+import { TelemetryRepository } from 'core/repositories/telemetry_payload'
+import { TYPES } from 'core/types'
+import { inject, injectable } from 'inversify'
+import ms from 'ms'
+import yn from 'yn'
+
+import { GhostService } from '..'
+import { JobService } from '../job-service'
+
+const debug = DEBUG('stats')
+const TELEMETRY_URL = process.TELEMETRY_URL
+
+@injectable()
+export abstract class TelemetryStats {
+  protected abstract url: string
+  protected abstract lock: string
+  private interval: number
+
+  constructor(
+    @inject(TYPES.GhostService) protected ghostService: GhostService,
+    @inject(TYPES.Database) protected database: Database,
+    @inject(TYPES.LicensingService) protected licenseService: LicensingService,
+    @inject(TYPES.JobService) private jobService: JobService,
+    @inject(TYPES.TelemetryRepository) private telemetryRepo: TelemetryRepository
+  ) {
+    this.interval = ms('1d')
+  }
+
+  public async start() {
+    await this.run(this.lock, this.interval, this.url)
+
+    setInterval(this.run.bind(this, this.lock, this.interval, this.url), this.interval)
+  }
+
+  protected abstract getStats()
+
+  protected async run(lockResource: string, interval: number, url: string) {
+    const lock = await this.jobService.acquireLock(lockResource, interval - ms('1 minute'))
+    if (lock) {
+      debug('Acquired lock')
+      const stats = await this.getStats()
+      await this.sendStats(url, stats)
+    }
+  }
+
+  private async sendStats(url: string, stats) {
+    debug('Sending stats: %o', stats)
+    try {
+      await axios.post(url, stats)
+    } catch (err) {
+      if (url === TELEMETRY_URL) {
```

Comment:
why not use straight up process.TELEMETRY_URL ?

Reply:
a208a8da8f104e184531ccdc8a17482c3bb38b4d